internal/debug/flags.go

```diff
@@ -55,6 +55,11 @@ var (
 		Name:  "debug",
 		Usage: "Prepends log messages with call-site location (file and line number)",
 	}
+	swaplogpathFlag = cli.StringFlag{
```

Comment:
Then, I suggest that when we read the flag (either from debug or from main), that this value is passed to the initialization of the `Swap` object (`New`), so that then we can set the `swaplogpath` there and we don't drag Swap-related stuff into this package, and all of the initialization of Swap stuff is in that package.

Reply:
> Then, I suggest that when we read the flag (either from debug or from main), that this value is passed to the initialization of the `Swap` object (`New`), so that then we can set the `swaplogpath` there and we don't drag Swap-related stuff into this package, and all of the initialization of Swap stuff is in that package.

So I would initialize it on the New Swap

Reply:
> First of all apologies for not noting this before.
> 
> I think this is not a debug flag, it is actually maybe a normal operation flag, so it should not be in the `debug` package.
> Move it to `cmd/swarm/flags.go`
> 
> I will ask opinion from @nonsense, @janos and @skylenet on this one

My opinion is it would be better as a debug flag, the amount of logs will be tremendous.
It could be used as an audit/debug trail.

Reply:
I understand your point of view and it is a good argument.

Audit usually denotes something which is legal. A bank always has to be able to show audit trails of its operations. However, we don't have any legal stuff here.

It is merely for the node operator to be able to monitor how his node is doing financially (profitability). We could maybe think of some tools (much) later which help analyze the file content. This is the reasoning for me to have this by default, not as a debug tool. 

I think we need more opinions - core and @vojtechsimetka for example.

Reply:
For legal reasons, financial transactions usually have to be kept for x years. I understand your concern with regards to the growth in file size, but I don't think there is a way around this.

If any node-operator will run Swarm for-profit, he most likely has to comply with regulation (whether he does this or not is his concern). Having this logger will allow him to do so.

If the purpose is merely to show the node operator how his node is doing business-wise, I think we can solve this better by making snapshots and store this to disk, especially because a log-file is probably not understandable (too big and too much jargon) for the average Swarm user.

Reply:
The flag indicates the path to swap related audit logs.
I can be used for debugging scenarios, or for auditing purposes.

I do agree on the legal side, but the level of year etc will depend on the legality of each country.

Personally I think the decision on storing the log files, should be carried out by the swarm operator.

Reply:
> First of all apologies for not noting this before.
> 
> I think this is not a debug flag, it is actually maybe a normal operation flag, so it should not be in the `debug` package.
> Move it to `cmd/swarm/flags.go`
> 
> I will ask opinion from @nonsense, @janos and @skylenet on this one

I moved the flag into `cmd/swarm/flags.go`.
It makes more sense this way.

I agree that we need more opinions.

Reply:
I personally think it is good now. Let's see if someone from core has an opinion.